source/multigrid/mg_transfer_internal.cc

```diff
@@ -443,6 +443,24 @@ namespace internal
               fe.get_prolongation_matrix(c)(renumbering[j],renumbering[i]);
     }
 
+    namespace
+    {
+      /**
+       * helper function for setup_transfer
+       */
+      void replace_constrainted_dof_entries(const ConstraintMatrix &level_constraints,
+                                            std::vector<types::global_dof_index> &dof_indices)
+      {
+        for (unsigned int i=0; i<dof_indices.size(); ++i)
```

Comment:
> Also i would iterate here as

Another point where I disagree. I find it a lot harder to understand that you are now modifying entries using a non-const iterator.